src/formats/bmp.c

```diff
@@ -383,11 +400,27 @@ enum loader_status decode_bmp(struct image* ctx, const uint8_t* data,
     color_data_sz = hdr->offset - sizeof(struct bmp_file) - bmp->dib_size;
     palette.table = color_data;
     palette.size = color_data_sz / sizeof(uint32_t);
-    mask = (color_data_sz <= sizeof(*mask) ? color_data : NULL);
+
+    // create mask
+    if (bmp->dib_size > BITMAPINFOHEADER_SIZE) {
+        mask_location = (const uint32_t*)(bmp + 1);
+
+	  mask.red = mask_location[0];
```

Comment:
Please, reformat the code with project's [.clang-format](https://github.com/artemsen/swayimg/blob/master/.clang-format), this will remove tab/space mix.